src/main/java/edu/harvard/iq/dataverse/util/FileUtil.java

```diff
@@ -1471,73 +1471,59 @@ public static String getCiteDataFileFilename(String fileTitle, FileCitationExten
             logger.fine("Download popup required because of guestbook.");
             return true;
         }
-
         logger.fine("Download popup is not required.");
         return false;
     }
     
-    public static boolean isRequestAccessPopupRequired(DatasetVersion datasetVersion){
-        // Each of these conditions is sufficient reason to have to 
-        // present the user with the popup: 
+    public static boolean isRequestAccessPopupRequired(DatasetVersion datasetVersion) {
+        
+        Boolean answer = popupDueToStateOrTerms(datasetVersion);
+        if (answer != null) {
+            return answer;
+        }
+        logger.fine("Request access popup is not required.");
+        return false;
+    }
+    
+    private static Boolean popupDueToStateOrTerms(DatasetVersion datasetVersion) {
+
+        // Each of these conditions is sufficient reason to have to
+        // present the user with the popup:
         if (datasetVersion == null) {
-            logger.fine("Download popup required because datasetVersion is null.");
+            logger.fine("Popup required because datasetVersion is null.");
             return false;
         }
-        //0. if version is draft then Popup "not required"
+        // 0. if version is draft then Popup "not required"
         if (!datasetVersion.isReleased()) {
-            logger.fine("Download popup required because datasetVersion has not been released.");
+            logger.fine("Popup required because datasetVersion has not been released.");
             return false;
         }
         // 1. License and Terms of Use:
         if (datasetVersion.getTermsOfUseAndAccess() != null) {
-            if (!datasetVersion.getTermsOfUseAndAccess().getLicense().isDefault()
-                    && !(datasetVersion.getTermsOfUseAndAccess().getTermsOfUse() == null
-                    || datasetVersion.getTermsOfUseAndAccess().getTermsOfUse().equals(""))) {
+            License license = datasetVersion.getTermsOfUseAndAccess().getLicense();
+            if ((license == null && StringUtils.isNotBlank(datasetVersion.getTermsOfUseAndAccess().getTermsOfUse()))
+                    || (license != null && !license.isDefault())) {
                 logger.fine("Download popup required because of license or terms of use.");
                 return true;
             }
 
             // 2. Terms of Access:
             if (!(datasetVersion.getTermsOfUseAndAccess().getTermsOfAccess() == null) && !datasetVersion.getTermsOfUseAndAccess().getTermsOfAccess().equals("")) {
-                logger.fine("Download popup required because of terms of access.");
+                logger.fine("Popup required because of terms of access.");
                 return true;
             }
         }
-
-        logger.fine("Download popup is not required.");
-        return false;
+        return null;
```

Comment:
Found the use of null here a little confusing, but looking at the rest of the code I see that it means "leave it up to Guestbook if applicable"

Reply:
Added a method header comment and a comment at this line.